src/main/java/com/aws/greengrass/componentmanager/ComponentStore.java

```diff
@@ -478,7 +486,8 @@ public RecipeMetadata getRecipeMetadata(ComponentIdentifier componentIdentifier)
 
             throw new PackageLoadingException(String.format(
                     "Failed to get recipe metadata because the file doesn't not exit or it is a folder. "
```

Comment:
file does not exist or it is a folder

we should be able to distinguish file doesn't exist from a folder though...

also remove the trailing "." - you can see it in a bunch of exceptions in here. We should be consistent and not have trailing "."